source/common/network/address_impl.cc

```diff
@@ -49,18 +30,16 @@ std::string friendlyNameFromAbstractPath(absl::string_view path) {
 const SocketInterface* sockInterfaceOrDefault(const SocketInterface* sock_interface) {
   return sock_interface == nullptr ? &SocketInterfaceSingleton::get() : sock_interface;
 }
-
 } // namespace
-
-Address::InstanceConstSharedPtr addressFromSockAddr(const sockaddr_storage& ss, socklen_t ss_len,
-                                                    bool v6only) {
+InstanceConstSharedPtr addressFromSockAddr(const sockaddr_storage& ss, socklen_t ss_len,
+                                           bool v6only) {
   RELEASE_ASSERT(ss_len == 0 || static_cast<unsigned int>(ss_len) >= sizeof(sa_family_t), "");
   switch (ss.ss_family) {
   case AF_INET: {
     RELEASE_ASSERT(ss_len == 0 || static_cast<unsigned int>(ss_len) == sizeof(sockaddr_in), "");
     const struct sockaddr_in* sin = reinterpret_cast<const struct sockaddr_in*>(&ss);
     ASSERT(AF_INET == sin->sin_family);
-    return std::make_shared<Address::Ipv4Instance>(sin);
+    return *createInstance<Address::Ipv4Instance>(sin);
```

Comment:
Should we propagate errors up by making addressFromSockAddr return a StatusOr, and drop the `*` here?

Or we can use a RELEASE_ASSERT(status_or.ok()) here and rename this method addressFromSockAddrOrDie.

Reply:
Thanks, I am working on it.

Reply:
We can return a StatusOr here.